test/libsolidity/syntaxTests/using/user_type_operator_defined_as_global.sol

```diff
@@ -0,0 +1,15 @@
+==== Source: s1.sol ====
+type Int is int;
+using {add as +} for Int global;
```

Comment:
Some more cases with `global`:
- File defining both global and non-global operators. Only global ones should work outside of it.
- File defining the same operator as global first, then redefining as non-global. Or the other way around. Which one wins?
- Unary and binary `-` defined in the same file, one global, the other not.
- Importing same global operator from two different files.
- Importing same operator from two files, one global, other not global.
- Having a global operator defined in one file but **not** importing it. Then importing a non-global operator from another file. Global stuff still must be imported to take effect, right?
- One file defines the operator as global, second file imports it and redefines as non-global, then the third file imports the second one. Does the third file see the operator or not? And which one?
- Global operators defined in a contract scope.

Reply:
> - File defining both global and non-global operators. Only global ones should work outside of it.

Test added.

> - File defining the same operator as global first, then redefining as non-global. Or the other way around. Which one wins?

No one wins since two different functions bound to the same operator cause compilation error regardless of global/non-global specification. I don't think it is necessary, but still I've added test cases that covers that case.

> - Unary and binary - defined in the same file, one global, the other not.

Test added.

> - Importing same global operator from two different files.
> - Importing same operator from two files, one global, other not global.

Taking into consideration global can be only used with types defined in the same source unit, it is not possible. But, I have created a test where I bind function to an operator to which function is already bound as global in imported file.

> - Having a global operator defined in one file but not importing it. Then importing a non-global operator from another file. Global stuff still must be imported to take effect, right?

Actually, `global` does not change the directive behavior in the current file. So, it doesn't need to be imported to take effect.

> - One file defines the operator as global, second file imports it and redefines as non-global, then the third file imports the second one. Does the third file see the operator or not? And which one?

There is no option to redefine operator so far. Whenever two functions are bound to one operator, compiler raise error. No matter if the function bound occurs more than once in the current file or is defined as global in an external file and then the file is imported.

> - Global operators defined in a contract scope.

Test added.

Reply:
> There is no option to redefine operator so far. Whenever two functions are bound to one operator, compiler raise error. No matter if the function bound occurs more than once in the current file or is defined as global in an external file and then the file is imported.

I'd still add that as a test case, even if it fails. This is actually the corner case that I was interested in the most here. The point is not just to see if it works but also to have a regression test so that we'll notice if behavior in these cases ever changes.